test/unit/org/apache/cassandra/sensors/DisabledSensorsTest.java

```diff
@@ -0,0 +1,224 @@
+import org.apache.cassandra.utils.ByteBufferUtil;
+import org.apache.cassandra.utils.UUIDGen;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests that sensors are not tracked when disabled. Please note that sensor tracking is and opt-in feature so any non-sensors related tests is indirectly leveraged and would surface
+ * functionality regression (e.g. NPEs). Here we make sure the sensors are indeed not tracked and for that it suffices to cover each verb handler that has sensor related implementation
+ * and asserting that indeed no sensors (of any type) are tracked. Ideally we would run all existing sensor tests with flag disabled but that would be an overkill.
+ *
+ */
+public class DisabledSensorsTest
+{
+    public static final String KEYSPACE1 = "SensorsReadTest";
+    public static final String CF_STANDARD = "Standard";
+    public static final String CF_STANDARD_SAI = "StandardSAI";
+    private static final String CF_COUTNER = "Counter";
+
+    private ColumnFamilyStore store;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception
+    {
+        CassandraRelevantProperties.REQUEST_SENSORS_FACTORY.setString(SensorsTestUtil.DefaultRequestSensorsFactory.class.getName());
+
+        // build SAI indexes
+        Indexes.Builder saiIndexes = Indexes.builder();
+        saiIndexes.add(IndexMetadata.fromSchemaMetadata(CF_STANDARD_SAI + "_val", IndexMetadata.Kind.CUSTOM, new HashMap<>()
+        {{
+            put(IndexTarget.CUSTOM_INDEX_OPTION_NAME, StorageAttachedIndex.class.getName());
+            put(IndexTarget.TARGET_OPTION_NAME, "val");
+        }}));
+
+        SchemaLoader.prepareServer();
+        SchemaLoader.createKeyspace(KEYSPACE1,
+                                    KeyspaceParams.simple(1),
+                                    SchemaLoader.standardCFMD(KEYSPACE1, CF_STANDARD,
+                                                              1, AsciiType.instance, AsciiType.instance, null),
+                                    SchemaLoader.standardCFMD(KEYSPACE1, CF_STANDARD_SAI,
+                                                              1, AsciiType.instance, AsciiType.instance, null)
+                                                .partitioner(Murmur3Partitioner.instance)
+                                                .indexes(saiIndexes.build()),
+                                    SchemaLoader.counterCFMD(KEYSPACE1, CF_COUTNER));
+
+        CompactionManager.instance.disableAutoCompaction();
+        DatabaseDescriptor.setPartitionerUnsafe(Murmur3Partitioner.instance);
+    }
+
+    @After
+    public void afterTest()
+    {
+        Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF_STANDARD).truncateBlocking();
+        Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF_STANDARD_SAI).truncateBlocking();
+        Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF_COUTNER).truncateBlocking();
+    }
+
+    @Test
+    public void testSensorsForReadVerbHandlerDisabled()
```

Comment:
nit: no need to suffix everything with "disabled" given this class is named `DisabledSensorsTest`.